portfolio/src/main/java/com/google/sps/servlets/AnalysisServlet.java

```diff
@@ -44,12 +40,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
         // Check if video is in database and retrieve it if it is.
         Key videoKey = KeyFactory.stringToKey(keyString);
         Entity videoEntity;
+        Entity videoAnalysisEntity;
         try {
             videoEntity = datastore.get(videoKey);
+            videoAnalysisEntity = datastore.get(videoEntity.getParent());
```

Comment:
Would be interested in hearing more about some of the design decisions you made. Normally, I would think of the Video as being the parent but it looks like here you made the analysis the parent. Is this because each entity can only have a single parent which means you can access the analysis easily from the video entity?

Reply:
Exactly what you said, I was going to have the analysis as the child but datastore only allows for direct reference to the parents of an entity (because entities can only have 1 parent but can have multiple children), so treating the video as a child allows us to toss around the part of the tree that would have taken longer to get to while being able to directly reference the analysis.